charmcraft/commands/store/client.py

```diff
@@ -126,6 +136,27 @@ def request(self, method, url, body):
         return resp
 
 
+def _storage_push(monitor):
+    """Push bytes to the storage."""
+    session = requests.Session()
+    retries = Retry(total=5, backoff_factor=2, status_forcelist=[500, 502, 503, 504])
+    session.mount("https://", HTTPAdapter(max_retries=retries))
+
+    url = STORAGE_BASE_URL + 'unscanned-upload/'
```

Comment:
We can not use `path.join` as it depends on the platform (it's for directories!).

The one I intended to use is `urllib.urljoin`, though, but has some weird semantics:

```
>>> parse.urljoin('http://foo.com/bar/baz/', 'unscanned-upload/')
'http://foo.com/bar/baz/unscanned-upload/'
>>> parse.urljoin('http://foo.com/bar/baz', 'unscanned-upload/')
'http://foo.com/bar/unscanned-upload/'
```

In any case, your worrying is valid. So I added a slash here and removed it in the base url:

```
STORAGE_BASE_URL = 'https://storage.staging.snapcraftcontent.com'
    url = STORAGE_BASE_URL + '/unscanned-upload/'
```

The worst thing it could happen is that somebody for testing uses a slash at the end of that base url (`http:localhost:8080/`), which would create an url with a double slash (`http:localhost:8080//unscanned-upload/`), which is valid anyway.

Reply:
hm. very strange behavior for urljoin. I think I was confused with Golang which has path.join() and filepath.join(), where the latter is platform specific and the former is always '/'.